junest: init at 7.4.8

jdev082 opened the pull request:
## Description of changes
**A new PR was made with a complete redo as git seemed to not want to cooperate**
[Original](https://github.com/NixOS/nixpkgs/pull/305418)

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

vinnymeller reviewed (approved):
looks good! :tada:

kirillrdy commented:
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

> When adding yourself as maintainer in the same pull request, make a separate commit with the message maintainers: add <handle>. Add the commit before those making changes to the package or module. See [Nixpkgs Maintainers](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) for details.

also, version in commit needs to match version of derivation ( drop 'v')

jdev082 commented:
> https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions
> 
> > When adding yourself as maintainer in the same pull request, make a separate commit with the message maintainers: add . Add the commit before those making changes to the package or module. See [Nixpkgs Maintainers](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) for details.
> 
> also, version in commit needs to match version of derivation ( drop 'v')

Fixed! I renamed the commit.

kirillrdy commented:
you can force push to this branch, same PR will be fine

kirillrdy commented on maintainers/maintainer-list.nix:
```diff
@@ -9217,6 +9217,12 @@
+  jdev082 = {
+    email = "jdev0894@gmail.com";
+    github = "jdev082";
+    githubId = 	92550746;
```
```suggestion
    githubId = 92550746;
```

jdev082 replied:
fixed!

jdev082 commented:
I have split the commit and put the two commits in the correct order.

kirillrdy commented:
not a big deal, but during setup I get 
```
/nix/store/57g7mn1x46ywczv140gsn1s64zry6pr6-junest-7.4.8/lib/core/common.sh: line 128: wget: command not found
```
it still works because it falls back to curl ( i wonder why curl is installed on my system )

kirillrdy commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,30 @@
+  meta = {
+    description = "Arch distro that runs on top of another without root";
+    homepage = "https://github.com/fsquillace/junest";
+    license = lib.licenses.gpl3;
```
https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

jdev082 replied:
Fixed!

jdev082 commented:
> not a big deal, but during setup I get
> 
> ```
> /nix/store/57g7mn1x46ywczv140gsn1s64zry6pr6-junest-7.4.8/lib/core/common.sh: line 128: wget: command not found
> ```
> 
> it still works because it falls back to curl ( i wonder why curl is installed on my system )

I've added wget as a buildInput so it doesn't fallback

kirillrdy commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,34 @@
+  dontBuild = true;
+
+  buildInputs = [
+    wget
```
wget is not a buildInput, it's a runtime dependency 

if you want to provide wget as runtime dependency you can use https://nixos.org/manual/nixpkgs/unstable/#fun-wrapProgram

but it's not strictly required, I'll leave it up to you

jdev082 replied:
Done!

kirillrdy commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,34 @@
+    cp -r $src/lib/ $out/
+  '';
+
+  postBuild = ''
```
because this is postBuild. it gets run before install, install overwrites `$out/bin/junest`

jdev082 replied:
Fixed!

kirillrdy commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,35 @@
+    mkdir -p $out/lib
+    cp -r $src/bin/ $out/
+    cp -r $src/lib/ $out/
+    wrapProgram $out/bin/junest --prefix PATH ${wget}/bin
```
wrapper still didnt work

```
cat ./results/junest/bin/junest 
#! /nix/store/dy2z01kpnxn7dn2kgfdxs4fm8xy9mb89-bash-5.2p26/bin/bash -e
exec -a "$0" "/nix/store/x4qb7mvhfi6xck6yyfqsi5b3r9d6q3s6-junest-7.4.8/bin/.junest-wrapped"  "$@" 
```

maybe you need `:` after PATH ?

jdev082 replied:
I built the package locally and can't find any problems. The wrapper generated works, and it seems to be using wget as intended. 

I ran:
```
nix build .#junest
```

kirillrdy replied:
whats the content of 
```
cat ./results/junest/bin/junest 
```

my doesn't have wget PATH set

jdev082 replied:
After further investigation it is not in fact working. However, I have found a fix.

jdev082 replied:
Should work now!

kirillrdy commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,36 @@
+    mkdir -p $out/lib
+    cp -r $src/bin/ $out/
+    cp -r $src/lib/ $out/
+    substituteInPlace $out/lib/core/common.sh --replace "wget" "${wget}/bin/wget"
```
```suggestion
    substituteInPlace $out/lib/core/common.sh --replace-fail "wget" "${wget}/bin/wget"
```

replace is deprecated

AndersonTorres commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,36 @@
+{ lib, stdenv, fetchFromGitHub, makeWrapper, wget }:
+
+stdenv.mkDerivation rec {
+  version = "7.4.8";
+  pname = "junest";
```
```suggestion
  pname = "junest";
  version = "7.4.8";

```

AndersonTorres commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,37 @@
+{ lib, stdenv, fetchFromGitHub, makeWrapper, wget }:
+
+stdenv.mkDerivation rec {
```
Is GCC used somehow?

kirillrdy replied:
good point, it's all bash

AndersonTorres commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,37 @@
+{ lib, stdenv, fetchFromGitHub, makeWrapper, wget }:
+
+stdenv.mkDerivation rec {
```
```suggestion
stdenvNoCC.mkDerivation rec {
```

AndersonTorres commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,37 @@
+
+  dontBuild = true;
+
+  buildInputs = [
+    wget
+  ];
```
```suggestion
```

AndersonTorres commented on pkgs/by-name/ju/junest/package.nix:
```diff
@@ -0,0 +1,37 @@
+    mkdir -p $out/lib
+    cp -r $src/bin/ $out/
+    cp -r $src/lib/ $out/
+    substituteInPlace $out/lib/core/common.sh --replace-fail "wget" "${wget}/bin/wget"
```
```suggestion
    substituteInPlace $out/lib/core/common.sh --replace-fail "wget" (lib.getExe wget)
```

kirillrdy commented:
i get eval error during nixpkgs review

```
nix log /nix/store/wc345ipf7gfhk7y04blyf7sx4isx9hl4-junest-7.4.8.drv
warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/wc345ipf7gfhk7y04blyf7sx4isx9hl4-junest-7.4.8.drv^*'
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/yrix7jhz9ni8rb2sfvm3phg39qj507m7-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: configurePhase
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: installPhase
/nix/store/rg6agzlb4gcl4w2126m16qm4mgr8jgi6-stdenv-linux/setup: eval: line 1583: syntax error near unexpected token `
```

Commit:
maintainers: add jdev082

Signed-off-by: jdev082 <jdev082@jdev.eu.org>

Commit:
junest: init at 7.4.8

Signed-off-by: jdev082 <jdev082@jdev.eu.org>

jdev082 commented:
> i get eval error during nixpkgs review
> 
> ```
> nix log /nix/store/wc345ipf7gfhk7y04blyf7sx4isx9hl4-junest-7.4.8.drv
> warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/wc345ipf7gfhk7y04blyf7sx4isx9hl4-junest-7.4.8.drv^*'
> @nix { "action": "setPhase", "phase": "unpackPhase" }
> Running phase: unpackPhase
> unpacking source archive /nix/store/yrix7jhz9ni8rb2sfvm3phg39qj507m7-source
> source root is source
> @nix { "action": "setPhase", "phase": "patchPhase" }
> Running phase: patchPhase
> @nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
> Running phase: updateAutotoolsGnuConfigScriptsPhase
> @nix { "action": "setPhase", "phase": "configurePhase" }
> Running phase: configurePhase
> no configure script, doing nothing
> @nix { "action": "setPhase", "phase": "installPhase" }
> Running phase: installPhase
> /nix/store/rg6agzlb4gcl4w2126m16qm4mgr8jgi6-stdenv-linux/setup: eval: line 1583: syntax error near unexpected token `
> ```

This error is caused by the following line:
```
    substituteInPlace $out/lib/core/common.sh --replace-fail "wget" (lib.getExe wget)
```
in the install phase.

> i get eval error during nixpkgs review
> 
> ```
> nix log /nix/store/wc345ipf7gfhk7y04blyf7sx4isx9hl4-junest-7.4.8.drv
> warning: The interpretation of store paths arguments ending in `.drv` recently changed. If this command is now failing try again with '/nix/store/wc345ipf7gfhk7y04blyf7sx4isx9hl4-junest-7.4.8.drv^*'
> @nix { "action": "setPhase", "phase": "unpackPhase" }
> Running phase: unpackPhase
> unpacking source archive /nix/store/yrix7jhz9ni8rb2sfvm3phg39qj507m7-source
> source root is source
> @nix { "action": "setPhase", "phase": "patchPhase" }
> Running phase: patchPhase
> @nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
> Running phase: updateAutotoolsGnuConfigScriptsPhase
> @nix { "action": "setPhase", "phase": "configurePhase" }
> Running phase: configurePhase
> no configure script, doing nothing
> @nix { "action": "setPhase", "phase": "installPhase" }
> Running phase: installPhase
> /nix/store/rg6agzlb4gcl4w2126m16qm4mgr8jgi6-stdenv-linux/setup: eval: line 1583: syntax error near unexpected token `
> ```

fixed!

kirillrdy reviewed (approved):
Result of `nixpkgs-review pr 306644` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>junest</li>
  </ul>
</details>

kirillrdy commented:
@jdev082 thank you !
